tests/system/general/test_45_FilterSolutions/test_distributed_filter/test_distributed_topology.py

```diff
@@ -0,0 +1,541 @@
+
+        with pytest.raises(RuntimeError) as info:
+            design.topology.pinned = True
+        assert info.value.args[0] == "The Pinned property is not supported for the selected topology"
+        design.attributes.filter_class = FilterClass.BAND_PASS
+        design.topology.topology_list = TopologyList.PARALLEL_EDGE_COUPLED
+        assert design.topology.pinned is False
+        design.topology.pinned = True
+        assert design.topology.pinned
+        assert design.topology.equal_width_conductors is False
+        assert design.topology.circuit_response().splitlines() == read_resource_file("pinned.ckt", "Distributed")
+
+    def test_stub_taps(self):
+        design = ansys.aedt.core.filtersolutions.DistributedDesign(config["desktopVersion"])
+
+        with pytest.raises(RuntimeError) as info:
+            design.topology.stub_taps = True
+        assert info.value.args[0] == "The Stub Taps property is not supported for the selected topology"
+        design.attributes.filter_class = FilterClass.BAND_PASS
+        design.topology.topology_list = TopologyList.PARALLEL_EDGE_COUPLED
+        assert design.topology.stub_taps is False
+        design.topology.stub_taps = True
+        assert design.topology.stub_taps
+        assert design.topology.circuit_response().splitlines() == read_resource_file("stub_taps.ckt", "Distributed")
+
+    def test_via_ends(self):
+        design = ansys.aedt.core.filtersolutions.DistributedDesign(config["desktopVersion"])
+
+        with pytest.raises(RuntimeError) as info:
+            design.topology.via_ends = True
+        assert info.value.args[0] == "The Via Ends property is not supported for the selected topology"
+        design.attributes.filter_class = FilterClass.BAND_PASS
+        design.topology.topology_list = TopologyList.PARALLEL_EDGE_COUPLED
+        assert design.topology.via_ends is False
+        design.topology.via_ends = True
+        assert design.topology.via_ends
+        assert design.topology.equal_width_conductors is False
+        assert design.topology.circuit_response().splitlines() == read_resource_file("via_ends.ckt", "Distributed")
+
+    def test_resonator_line_width(self):
+        design = ansys.aedt.core.filtersolutions.DistributedDesign(config["desktopVersion"])
+
+        with pytest.raises(RuntimeError) as info:
+            design.topology.resonator_line_width = "1.27 mm"
+        assert info.value.args[0] == "The Line Width property is not supported for the selected topology"
+        design.attributes.filter_class = FilterClass.BAND_PASS
+        design.topology.topology_list = TopologyList.HAIRPIN
+        assert design.topology.resonator_line_width == "1.27 mm"
+        design.topology.resonator_line_width = "5 mm"
+        assert design.topology.resonator_line_width == "5 mm"
+        assert design.topology.circuit_response().splitlines() == read_resource_file(
+            "resonator_line_width.ckt", "Distributed"
```

Comment:
resonator_line_width_5mm.ckt

I will stop commenting about this, but please review the file names. This will help make maintenance much easier.